### ISSUE TITLE
Feature Request: Permission inheritance

### ISSUE THREAD

**Issue by michael-soorys** [hasura/graphql-engine#6591]
Hi Hasura team, really love using your product and would like to give some user feedback. I am not sure if this was brought up before, but I really wish that this was a feature. 

### The problem:
- Table A has complex permissions in place based on multiple tables and relationships.
- I need to base the permission level of table B, C, and D on the permissions of table A. Table A has a direct one to many relationship with B, C and D.

Basically if a user can get to table A then he can get to table B, C and D, else nothing is available. Like an operator in the permission dropdown options "B" -> "A" -> "_can_access"

**Comment by leoalves** [comment]
There is an RFC to disable root level fields.  
This might be the solution for your need.
https://github.com/0x777/graphql-engine/blob/select-permission-improvements/rfcs/disable-query-and-subscription-root-fields.md

But I don't know if the Hasura team has an ETA on this RFC.

**Comment by afitzek** [comment]
I think you want something similar than what I suggested here: https://github.com/hasura/graphql-engine/issues/6226 :)

**Comment by Gregoor** [comment]
We have that exact same need. I think an abstraction mechanism at the permission level is essential, otherwise two bad things happen:
- repetition which eventually leads to inconsistencies, meaning in this case security issues
- confused people like me will use SQL views to build "access"-tables which will seem like a nice abstraction but will turn out to horribly degrade performance because it kills Hasura's amazing predicate+relationship bundling optimizations
